File: 2132-convert-1d-array-into-2d-array/convert-1d-array-into-2d-array.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        # using slicing logic
        res = []
        if len(original) == m*n:
            for i in range(0, len(original), n):
                res.append(original[i:n+i])
        return res

File: 2132-convert-1d-array-into-2d-array/convert-1d-array-into-2d-array.java
class Solution {
    public int[][] construct2DArray(int[] original, int m, int n) {
        if (original.length != m * n) {
            return new int[0][0]; 
        }    

        int row = 0; 
        int col = 0;

        int[][] arr = new int[m][n];

        for (int i = 0; i < original.length; i++){
            arr[row][col] = original[i];
            col++;
            if (col == n){
                col = 0;
                row++;
            }
        }
        return arr;
    }
}